test_conformance/api/test_api_consistency.cpp

```diff
@@ -1148,3 +1148,124 @@ int test_consistency_3d_image_writes(cl_device_id deviceID, cl_context context,
 
     return TEST_PASS;
 }
+
+int test_consistency_requirements_fp64(cl_device_id deviceID,
+                                       cl_context context,
+                                       cl_command_queue queue, int num_elements)
+{
+    cl_int error = CL_SUCCESS;
+    cl_device_fp_config value = 0;
+
+    if (is_extension_available(deviceID, "cl_khr_fp64"))
+    {
+        error = clGetDeviceInfo(deviceID, CL_DEVICE_DOUBLE_FP_CONFIG,
+                                sizeof(value), &value, nullptr);
+        test_error(error, "Unable to get device CL_DEVICE_DOUBLE_FP_CONFIG");
+        test_assert_error(
+            value > 0, "CL_DEVICE_DOUBLE_FP_CONFIG must return nonzero value");
```

Comment:
Maybe a question for other reviewers or @bashbaug: the specification states some minimum double precision floating-point capabilities that we could check here instead of only checking that `value > 0`.  Would that be worth adding?
(and likewise for half precision)

Reply:
I agree - it would be preferable to check for the minimum set of required values.